FreeRTOS-Plus/Source/Application-Protocols/network_transport/mbedtls_pk_pkcs11.c

```diff
@@ -442,6 +460,7 @@ static CK_RV p11_ecdsa_ctx_init( void * pvCtx,
     else
     {
         xResult = CKR_FUNCTION_FAILED;
+        return xResult;
```

Comment:
I went with the early return because I thought it was cleaner than needing to wrap the rest of the function in the else block?

Where to be fair, this is a MISRA violation but I don't believe we enforce MISRA compliance on these files. But I can do that if you want.

There are many other places that the rules from MISRA are ignored here i.e. almost all of the casting that is done between [different types](https://github.com/FreeRTOS/FreeRTOS/blob/34b2048819a2fd85830e1489a36786ddc603f085/FreeRTOS-Plus/Source/Application-Protocols/network_transport/mbedtls_pk_pkcs11.c#L449):
```
P11EcDsaCtx_t * pxP11EcDsaCtx = ( P11EcDsaCtx_t * ) pk;
```

And the fact that a majority of the function parameters aren't marked `const` when they aren't modified. Where that's because MbedTLS expects these to be marked as non-constant, where making them constant would raise the dropping the `const` tag [compilation warning](https://learn.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-3-c4287?view=msvc-170)

Reply:
> Just musing: With this change, do we still need bigger timeouts?

I know that this could be reverted - I had mainly done this change for two reasons
1. I was originally seeing an issue with the task being starved/delayed when I was using the logging task to print out mbedtls debug info. Where doing this fixed that.
1. But then I thought it felt odd that we were showing examples of every task running at the same priority like this? This just seems more correct in my mind that this task *IS* a higher priority than the logging or idle task.
    * Where I had actually considered adding task priority changes with the other demos.
    * But the scope of this work has already gone well over what was intended, so I didn't do this.

Where all said, I'm fine with reverting this. But I'd prefer to leave it in, since it is inconsistent now.

Reply:
I don't mind the priority change and the timeout change. Lets keep it.

For the early return, I agree that these files are not checked for MISRA compliance but these files should be example of best practices. I also agree that adding else would make the code a bit different, but IMO, it is easier to debug if the function doesn't have a ton of return locations. As far as casting is considered, we cannot do much in those cases except add MISRA justifications saying why are we casting and why is it okay to do so.

Unless you are completely against it and willing to share the reasons, I propose that we stick with single point return.

Reply:
> Unless you are completely against it and willing to share the reasons, I propose that we stick with single point return.

Only reasoning I have is that I think since we ignore a lot of the MISRA best practices in the file, it feels odd to pick and choose which ones to follow? Where we are also ignoring the whole "if statements should have a corresponding else statement" in this function already.

But I agree with your logic that "When possible stick to best practices" so I went ahead and swapped to the single exit point for this function again